lib/read_preference.js

```diff
@@ -15,20 +15,11 @@ const ReadPreference = function(mode, tags, options) {
   if (!ReadPreference.isValid(mode)) {
     throw new TypeError(`Invalid read preference mode ${mode}`);
   }
-
-  // TODO(major): tags MUST be an array of tagsets
-  if (tags && !Array.isArray(tags)) {
-    console.warn(
-      'ReadPreference tags must be an array, this will change in the next major version'
-    );
-
-    if (typeof tags.maxStalenessSeconds !== 'undefined') {
-      // this is likely an options object
-      options = tags;
-      tags = undefined;
-    } else {
-      tags = [tags];
-    }
+  if (options === undefined && typeof tags === 'object' && !Array.isArray(tags)) {
```

Comment:
Not sure how much effort we're trying to ensure this is an array.  I know this sucks...

```
> var tags = null
undefined
> typeof tags === 'object'
true
```

Reply:
I think null tags should be fine - it would fail Array.isArray and get passed through as null options, which is allowed (it'll default to an empty object).

Reply:
👍